### PR TITLE
Adding analytics docs

### PR DESCRIPTION
**Pull request by frehner**
Fixes: https://github.com/Shopify/hydrogen-react/issues/139

# Breaking Changes

* `useShopifyCookies` - if `hasUserConsent` is `false`, no cookies will be set
* `sendShopifyAnalytics` - if `hasUserConsent` is `false`, no analytics will be sent
* `ShopifyAppSource ` got rename to `ShopifySalesChannel `
* `getClientBrowserParameters` returns empty string for each field key if run on server

### CODE REVIEW

File: packages/react/docs/generated/generated_docs_data.json [hunk 1]
```diff
@@ -4716,6 +4716,292 @@
       }
     ]
   },
+  {
```
Comment (frehner) [inline packages/react/docs/generated/generated_docs_data.json:4719]: This is all auto-generated so no need to review this file.

File: packages/react/src/analytics.doc.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import {ReferenceEntityTemplateSchema} from '@shopify/generate-docs';
+
+const data: ReferenceEntityTemplateSchema = {
+  name: 'sendShopifyAnalytics',
+  category: 'utilities',
+  isVisualComponent: false,
+  related: [],
```
Comment (frehner) [inline packages/react/src/analytics.doc.ts:7]: We will probably want to add to this `related` array as we add other analytics docs

File: packages/react/src/analytics-constants.doc.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import {ReferenceEntityTemplateSchema} from '@shopify/generate-docs';
+
+const data: ReferenceEntityTemplateSchema = {
+  name: 'Analytics constants',
+  category: 'utilities',
+  isVisualComponent: false,
+  related: [],
```
Comment (frehner) [inline packages/react/src/analytics-constants.doc.ts]: Do you want to add some links to this related area, to the other analytics docs?

File: packages/react/src/analytics-types.ts [hunk 4]
```diff
@@ -27,37 +40,107 @@ export type ShopifyAnalyticsProduct = {
...
-export type ShopifyAppSources = keyof typeof ShopifyAppSource;
+export type ShopifySalesChannels = keyof typeof ShopifySalesChannel;
 export type AnalyticsEventNames = keyof typeof AnalyticsEventName;
 
-export type ShopifyCommonPayload = ClientBrowserParameters & {
-  shopifyAppSource?: ShopifyAppSources;
+export type ShopifyPageViewPayload = {
+  /** If we have consent from buyer for data collection */
   hasUserConsent: boolean;
+  /** Shopify shop id in the form of `gid://shopify/Shop/<id>` */
```
[the remark this answers is not in the document]
Comment (wizardlyhel) [reply packages/react/src/analytics-types.ts:49]: the whole of `gid` string

File: packages/react/src/analytics-types.ts [hunk 5]
```diff
@@ -27,37 +40,107 @@ export type ShopifyAnalyticsProduct = {
...
   shopId: string;
+  /** Currency code */
   currency: string;
```
[the remark this answers is not in the document]
Comment (wizardlyhel) [reply packages/react/src/analytics-types.ts:52]: yes it should

**Pull request merged**